akka-docs/src/main/paradox/typed/style-guide.md

```diff
@@ -190,3 +190,108 @@ Scala
 :  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #fun-style-setup-params4 }
 
 @@@
+
+## Behavior factory method
+
+The initial behavior should be created via @scala[a factory method in the companion object]@java[a static factory method].
+Thereby the usage of the behavior doesn't change when the implementation is changed, for example if
+changing between object-oriented and function style.
+
+The factory method is a good place for retrieving resources like `Behaviors.withTimers` and `Behaviors.withStash`.
+
+When using the object-oriented style it's a good convention to always create the instance from a `Behaviors.setup`
+block in the this factory method even though the `ActorContext` is not needed. The reason is that it prevents
+mistakes of sharing the same instance of the `Behavior` between different actor instances.
```

Comment:
It's true that it's not avoiding, but maybe reducing?

E.g. this would be wrong
```scala
val theBehavior = new MyBehavior(param)
context.spawn(theBehavior, "a")
context.spawn(theBehavior, "b")
```

but with `setup` in the factory method this would be ok:
```scala
val theBehavior = MyBehavior(param)
context.spawn(theBehavior, "a")
context.spawn(theBehavior, "b")
```

although we should prefer:
```scala
context.spawn(MyBehavior(param), "a")
context.spawn(MyBehavior(param), "b")
```

The first example is also wrong from a restart aspect.

That said, there are other ways to share in wrong way that are more likely so it's not a replacement for not understanding the basics of actor concurrency model. E.g. sharing a mutable parameter that is passed to several actors.

Reply:
@johanandren I think this might be even more important because of restart supervision. I rephrased it in e6bf13e

Thinking about if we should even enforce it (somewhat) by adding ActorContext as a constructor parameter in `AbstractBehavior`.

Reply:
Ah, good point about the caller sharing the returned value, didn't think of that.